spec/legacy/openstack_handle/handle_spec.rb

```diff
@@ -49,62 +49,60 @@
       auth_url = OpenstackHandle::Handle.auth_url("address", 5000, "https")
 
       expect(OpenstackHandle::Handle).to receive(:raw_connect).with(
-                                           "dummy",
-                                           "dummy",
-                                           "https://address:5000/v2.0/tokens",
-                                           "Compute",
-                                           {:openstack_tenant       => "admin",
-                                            :openstack_project_name => "admin",
-                                            :openstack_project_domain_id => nil,
-                                            :openstack_user_domain_id    => nil,
-                                            :openstack_region       => nil,
-                                            :connection_options     => {:ssl_verify_peer => false}})
-                                           .once do |_, _, address|
+        "dummy",
+        "dummy",
+        "https://address:5000",
+        "Compute",
+        :openstack_project_name      => "admin",
+        :openstack_project_domain_id => nil,
+        :openstack_user_domain_id    => nil,
+        :openstack_region            => nil,
+        :connection_options          => {:ssl_verify_peer => false}
+      ).once do |_, _, address|
         expect(address).to eq(auth_url)
         fog
       end
-      expect(handle.connect(:tenant_name => "admin")).to eq(fog)
+      expect(handle.connect(:openstack_project_name => "admin")).to eq(fog)
```

Comment:
I might not be tracing the logic correctly, but it looks like passing `openstack_project_name` here doesn't quite make sense. It looks like `connect` sees the `tenant_name` field is empty, but the `discover_tenants` option defaults to true so it sets `tenant = default_tenant_name`. When it gets to the conditional starting on line 166, it sees `tenant` is set so it uses that value, overwriting the value of `openstack_project_name` you passed in.